modules/agar/src/main/ts/ephox/agar/api/RealMouse.ts

```diff
@@ -58,6 +58,15 @@ const cClick = (): Chain<SugarElement<Element>, SugarElement<Element>> =>
 const pClickOn = (selector: string): Promise<{}> =>
   pActionOn(selector, 'click');
 
+const pClick = (elem: SugarElement<Element>): Promise<{}> => {
+  const id = Id.generate('');
+  Attribute.set(elem, BedrockIdAttribute, id);
+  const selector = `[${BedrockIdAttribute}="${id}"]`;
+  return pClickOn(selector).finally(() => {
+    Attribute.remove(elem, BedrockIdAttribute);
+  });
```

Comment:
This isn't a very async/await style... I'd suggest this instead:
```suggestion
  try {
    return pClickOn(selector);
  } finally {
    Attribute.remove(elem, BedrockIdAttribute);
  }
```